faunadb-java-dsl/src/main/java/com/faunadb/client/types/Field.java

```diff
@@ -106,6 +106,38 @@ public CollectionCodec(Path path, Field<A> field) {
     return new Field<>(Path.empty(), codec);
   }
 
+  /**
+   * Creates a field that converts its underyling value to the {@link Class} provided.
+   *
+   * @param <T> the desired final type
+   * @return a new {@link Field} instance
+   */
+  public static <T> Field<T> as(Class<T> clazz) {
+    return new Field<>(Path.empty(), new Codec<T>() {
+            @Override
+            public Result<T> decode(Value value) {
+              return Decoder.decode(value, clazz);
+            }
+
+            @Override
+            public Result<Value> encode(T value) {
+              return Encoder.encode(value);
+            }
+        });
+  }
+
+  /**
+   * Assuming the {@link Value} instance is a collection, creates a field extractor that collects
+   * the {@link Field} provided for each element in the underlying collection.
+   *
+   * @param <A> the desired final type for each collection element
+   * @param field the {@link Field} to be extracted from each collection element
+   * @return a new {@link Field} instance
+   */
+  public static <A> Field<List<A>> asListOf(Field<A> field) {
+    return new Field<>(Path.empty(), new CollectionCodec<>(Path.empty(), field));
+  }
+
```

Comment:
I think we also need a method to convert to `Map<String, T>`.

Reply:
Is there a Codec laying around for that already?

Reply:
`Codec.OBJECT`

Reply:
It compiles... so it must work, right? :)

Reply:
SureThing ™